src/main/webapp/app/shared/validators/custom-pattern-validator.directive.ts

```diff
@@ -0,0 +1,25 @@
+import { Directive } from '@angular/core';
+import { AbstractControl, NG_VALIDATORS, Validator } from '@angular/forms';
+
+/**
+ * Custom validator for patterns (RegEx)
+ */
+@Directive({
+    // tslint:disable-next-line:directive-selector
+    selector: '[validPattern][ngModel],[validPattern][formControl]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: CustomPatternValidatorDirective, multi: true }],
+})
+export class CustomPatternValidatorDirective implements Validator {
+    validate(control: AbstractControl): { [key: string]: any } | null {
+        if (control === undefined || control === null) {
```

Comment:
```suggestion
        if (control == undefined) {
```
That checks for both null and undefined at the same time.